[git-webkit] Add pickable command

JonWBedard opened the pull request:
#### 2f625033c95064c6f5a2a99c21be512eb0305bae
<pre>
[git-webkit] Add pickable command
<a href="https://bugs.webkit.org/show_bug.cgi?id=244296">https://bugs.webkit.org/show_bug.cgi?id=244296</a>
&lt;rdar://97398141&gt;

Reviewed by NOBODY (OOPS!).

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py: Add &apos;Pickable&apos; command.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/find.py:
(Info.print): Create function for printing a commit series.
(Info.main): Move printing into seperate function.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py: Added.
(Pickable.parser):
(Pickable.pickable): Return a list of cherry-pickable commits.
(Pickable.main): Find a list of commits, filter that list with the &apos;pickable&apos; function and print the result.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pickable_unittest.py: Added.
(TestPickable.setUp):
(TestPickable.test_none):
(TestPickable.test_main):
(TestPickable.test_main_git_svn):
(TestPickable.test_branch):
(TestPickable.test_branch_none):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2f625033c95064c6f5a2a99c21be512eb0305bae

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87012 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31098 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/17858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/95899 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/149592 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/90990 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29463 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/91042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92628 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/23784 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/73838 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/79141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/90619 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/78772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/17858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27195 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/17858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/86114 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27138 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/17858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28820 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/73838 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1060 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28761 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/17858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

dewei-zhu commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/find.py:
```diff
@@ -51,40 +51,12 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def main(cls, args, repository, reference='HEAD', **kwargs):
```
This is just a location move, right?

JonWBedard replied:
Yes! We're breaking out all of the `print` details and `git` makes this diff a bit weird.

dewei-zhu reviewed (approved):
r=me

JonWBedard committed:
[git-webkit] Add pickable command
https://bugs.webkit.org/show_bug.cgi?id=244296
<rdar://97398141>

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py: Add 'Pickable' command.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/find.py:
(Info.print): Create function for printing a commit series.
(Info.main): Move printing into seperate function.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py: Added.
(Pickable.parser):
(Pickable.pickable): Return a list of cherry-pickable commits.
(Pickable.main): Find a list of commits, filter that list with the 'pickable' function and print the result.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pickable_unittest.py: Added.
(TestPickable.setUp):
(TestPickable.test_none):
(TestPickable.test_main):
(TestPickable.test_main_git_svn):
(TestPickable.test_branch):
(TestPickable.test_branch_none):

Canonical link: https://commits.webkit.org/253761@main
